src/devices/Is31fl3730/Is31fl3730.cs

```diff
@@ -0,0 +1,423 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Device.I2c;
+using static System.Linq.Enumerable;
+
+namespace Iot.Device.Display
+{
+    /// <summary>
+    /// Represents an IS31FL3731 LED Matrix driver
+    /// </summary>
+    // Datasheet: https://cdn-shop.adafruit.com/product-files/3017/31FL3730.pdf
+    // Product: https://shop.pimoroni.com/products/microdot-phat
+    // Product: https://shop.pimoroni.com/products/led-dot-matrix-breakout
+    // Related repo: https://github.com/pimoroni/microdot-phat
+    // Based on: https://github.com/pimoroni/microdot-phat/blob/master/library/microdotphat/matrix.py
+    public class Is31fl3730
+    {
+        private readonly byte[] _matrix_addresses = new byte[] { FunctionRegister.Matrix1, FunctionRegister.Matrix2 };
+        private readonly List<byte[]> _buffers = new List<byte[]>
+        {
+            new byte[8],
+            new byte[8]
+        };
+        private readonly bool[] _enabled = new bool[2];
+        private I2cDevice _i2cDevice;
+        private int _configurationValue = 0;
+
+        /// <summary>
+        /// Initialize IS31FL3730 device
```

Comment:
I give the comment here but it's better to finish all comments with a dot. Some are finished properly like this, some are not. The linter isn't configured to force it but I have to say, I have a preference for sentences :-) So it's optional but still consistency is a good way to go.